why-polars-for-data-cleaning/index.qmd

```diff
@@ -0,0 +1,202 @@
+---
+title: "Why use Polars for data cleaning?"
+description: "The key to useful and meaningful data is data that is good quality and tidy. Getting data to be tidy and higher quality is an often undervalued and underestimated activity, at the same time being very labour intensive. But it is essential to have higher quality and tidier data to have better quality results from the analysis. This post explains why we chose to use Polars for data cleaning and processing tasks."
+date: "2024-12-04"
+categories:
+- data cleaning
+- pre-processing data
+---
+
+::: content-hidden
+Use other decision posts as inspiration to writing these. Leave the
+content-hidden sections in the text for future reference.
+:::
+
+## Context and problem statement
+
+::: content-hidden
+State the context and some background on the issue, then write a
+statement in the form of a question for the problem.
+:::
+
+When working with data analysis, data is rarely ready to be used 'as
+is'. There will always be things that are either very wrong (like text
+in what should be a number field), slightly wrong (like a single missing
+or incorrect value), or an outlier potentially due to an error in data
+entry. Since we don't have domain knowledge to know about missing values
+or the definition or meaning of an outlier. But there are some basic
+things we can check and fix, such as whether the data contradicts any
+constraints in the definition of the data resource (eg. a number in a
+date field). There are a lot of tools that can be used to both analyse a
+data set, and to clean it, either as part of the analysis, or as part of
+the initial check before the data is transformed and stored. So, our
+question is:
+
+*Which tool should we use for cleaning and processing data in our
+software and what tool should we recommend users to use when cleaning
+and processing needs to happen outside of our software?*
+
+## Decision drivers
+
+::: content-hidden
+List some reasons for why we need to make this decision and what things
+have arisen that impact work.
+:::
+
+-   Can be installed and used in Python.
+-   Is designed and built to be fast for processing data.
+-   Has minimal dependencies, so can be installed quickly.
+-   Is flexible to reading different data storage formats (e.g. SQL
+    databases, Parquet, CSV).
+-   Can write to different storage formats (as with above).
+-   We need a tool that can clean data and is either written in Python
+    or has a Python package available.
```

Comment:
```suggestion
```

I guess the re-wording now is nearly a duplicate of the first item and the third item, so can be removed now.